### ISSUE TITLE
Some annotation attribute cannot be resolve when enable annotation-based null analysis

### ISSUE THREAD

**Issue by testforstephen** [eclipse-jdt/eclipse.jdt.core#476]
Here is a sample project:   [annotationTest.zip](https://github.com/eclipse-jdt/eclipse.jdt.core/files/9799189/annotationTest.zip)
And it contains three classes.

- **RequestMapping.java**
```java
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface RequestMapping {
	String name() default "";
	String[] value() default {};
}
```

- **Controller.java**
```java
public class Controller<T> {
    final static String ENDPOINT = "controll";
}
```

- **CtlImpl.java**
```java
@RequestMapping(CtlImpl.CTL_ENDPOINT)
public class CtlImpl extends Controller<String> {
    final static String CTL_ENDPOINT = ENDPOINT + "/ctl";
    static String value;
}
```

**Reproduce Steps:**
- Open the sample project `annotationTest` in Eclipse
- Right click on this project from Package Explorer, and find "**Properties -> Java Compiler -> Error/Warnings -> Null analysis**", then check the box of "**enable annotation-based null analysis**"
- The build will fail with the following errors.
![image](https://user-images.githubusercontent.com/14052197/196148447-088ce548-2a86-4747-914f-605a7f9cdec9.png)

And if I disable the preference of "annotation-based null analysis", then the project works again.

**Comment by stephan-herrmann** [comment]
This will be fixed with #1077, might want to add this test case to that PR.

**Comment by stephan-herrmann** [comment]
Resolved via #1077
